plugin/helpers.go

```diff
@@ -37,8 +44,36 @@ type Helpers interface {
 	//
 	// Minimum server version: 5.6
 	KVSetWithExpiryJSON(key string, value interface{}, expireInSeconds int64) error
-}
 
+	InstallPluginFromUrl(url string, replace bool) (*model.Manifest, *model.AppError)
+}
 type HelpersImpl struct {
-	API API
+	API         API
+	HTTPService httpservice.HTTPService
```

Comment:
I guided you in the wrong direction. Using `httpservice.HTTPService` is no the right thing, as creating it is a good as using a `http.Client`. I think you can simplify this by just using `http.Client`.


In the tests you can just run use an [`http.Server`](https://golang.org/pkg/net/http/#Server) as the endpoint to make requests to.


Sorry for the misguidance.

Reply:
No problem. I did the same thing earlier. But I have concerns. Please have a look at https://github.com/mattermost/mattermost-server/blob/a2adf7b3f5ee106ea81a8996c10e026d87e07e34/services/httpservice/httpservice.go#L48-L60  We are not using default transport but our very own. So to make it consistent I tried using the `HTTPService`.

Reply:
@hanzei Please help me with the above query.

Reply:
Plugins are free to ignore settings for the Server as they are more "stand alone" programs. I don't think we need to respect `.ServiceSettings.EnableInsecureOutgoingConnections` here.

Reply:
I think we will require a simple `http.Client` then. I am making changes.